electrumsv/network_support/esv_client.py

```diff
@@ -0,0 +1,415 @@
+import aiohttp
+from aiohttp import web, WSServerHandshakeError
+import asyncio
+import base64
+import json
+from typing import List, Union, Optional, AsyncIterable, Dict, cast
+
+from bitcoinx import hash_to_hex_str
+
+from electrumsv.bitcoin import TSCMerkleProof
+from electrumsv.logs import logs
+from electrumsv.network_support.esv_client_types import (PeerChannelToken, TokenPermissions,
+    MessageViewModelGetBinary, GenericJSON, MessageViewModelGetJSON, APITokenViewModelGet,
+    PeerChannelViewModelGet, RetentionViewModel, TipResponse, Error, GeneralNotification, ChannelId,
+    WebsocketUnauthorizedException, PeerChannelMessage, MAPICallbackResponse, WebsocketError)
+
+logger = logs.get_logger("esv-client")
+
+REGTEST_MASTER_TOKEN = "t80Dp_dIk1kqkHK3P9R5cpDf67JfmNixNscexEYG0_xa" \
+                       "CbYXKGNm4V_2HKr68ES5bytZ8F19IS0XbJlq41accQ=="
+
+
+class PeerChannel:
+    """Represents a single Peer Channel instance"""
+
+    def __init__(self, channel_id: str, tokens: List[PeerChannelToken], base_url: str,
+            session: aiohttp.ClientSession, master_token: str) -> None:
+        assert len(base64.urlsafe_b64decode(channel_id)) == 64
+        for permissions, api_key in tokens:
+            assert len(base64.urlsafe_b64decode(api_key)) == 64
+        self.channel_id = channel_id
+        self.tokens = tokens
+        self.base_url = base_url
+        self.session = session
+        self.master_token = master_token  # master bearer token for the server account
+
+    def __repr__(self) -> str:
+        return f"<PeerChannel channel_id={self.channel_id}/>"
+
+    def get_callback_url(self) -> str:
+        return self.base_url + f"api/v1/channel/{self.channel_id}"
+
+    def get_write_token(self) -> PeerChannelToken:
```

Comment:
Get rid of the exception and return None if does not exist. We use that pattern as a standard approach everywhere and should stick to it.